control-plane/gateways/deployment_test.go

```diff
@@ -64,7 +64,7 @@ func Test_meshGatewayBuilder_Deployment(t *testing.T) {
 							constants.AnnotationGatewayWANAddress: "",
 						},
 					},
-					Spec: pbmesh.MeshGateway{
+					Spec: pbmesh.MeshGatewaySpec{
```

Comment:
I think these tests will need to be updated based on the changes in https://github.com/hashicorp/consul/pull/20159